src/libdrakvuf/linux-processes.c

```diff
@@ -267,3 +267,68 @@ bool linux_get_current_thread_id( drakvuf_t drakvuf, uint64_t vcpu_id, uint32_t
 
     return true;
 }
+
+vmi_pid_t linux_get_process_ppid( drakvuf_t drakvuf, addr_t process_base )
+{
+    uint32_t ppid = -1 ;
+    status_t ret = VMI_FAILURE ;
+    addr_t parent_proc_base = 0 ;
+    access_context_t ctx = {
+        .translate_mechanism = VMI_TM_PROCESS_PID,
+        .pid = 0,
+        .addr = process_base + drakvuf->offsets[TASK_STRUCT_REALPARENT] };
+
+    ret = vmi_read_addr( drakvuf->vmi, &ctx, &parent_proc_base );
```

Comment:
...just compare the function's return to VMI_FAILURE`

Reply:
If we finally redefine this function the way you're proposing I'm going to change substantially the coding

Reply:
Well... I'm afraid it not changed too much... :)